tests/functional/services/policy_engine/test_query_by_vulnerability.py

```diff
@@ -0,0 +1,153 @@
+)
+
+
+class TestQueryByVulnerability:
+    @pytest.mark.parametrize(
+        "query",
+        [
+            ImagesByVulnerabilityQuery(
+                "CVE-2013-2512",
+                [
+                    "sha256:80a31c3ce2e99c3691c27ac3b1753163214494e9b2ca07bfdccf29a5cca2bfbe",
+                    "sha256:406413437f26223183d133ccc7186f24c827729e1b21adc7330dd43fcdc030b3",
+                    "sha256:fe3ca35038008b0eac0fa4e686bd072c9430000ab7d7853001bde5f5b8ccf60c",
+                ],
+            ),
+            ImagesByVulnerabilityQuery(
+                "GHSA-h6q6-9hqw-rwfv",
+                [
+                    "sha256:80a31c3ce2e99c3691c27ac3b1753163214494e9b2ca07bfdccf29a5cca2bfbe",
+                    "sha256:406413437f26223183d133ccc7186f24c827729e1b21adc7330dd43fcdc030b3",
+                    "sha256:fe3ca35038008b0eac0fa4e686bd072c9430000ab7d7853001bde5f5b8ccf60c",
+                ],
+            ),
+            ImagesByVulnerabilityQuery(
+                "CVE-2021-21330",
+                [
+                    "sha256:80a31c3ce2e99c3691c27ac3b1753163214494e9b2ca07bfdccf29a5cca2bfbe",
+                    "sha256:406413437f26223183d133ccc7186f24c827729e1b21adc7330dd43fcdc030b3",
+                    "sha256:fe3ca35038008b0eac0fa4e686bd072c9430000ab7d7853001bde5f5b8ccf60c",
+                ],
+            ),
+            ImagesByVulnerabilityQuery(
+                "CVE-2020-8908",
+                [
+                    "sha256:80a31c3ce2e99c3691c27ac3b1753163214494e9b2ca07bfdccf29a5cca2bfbe",
+                    "sha256:406413437f26223183d133ccc7186f24c827729e1b21adc7330dd43fcdc030b3",
+                    "sha256:fe3ca35038008b0eac0fa4e686bd072c9430000ab7d7853001bde5f5b8ccf60c",
+                ],
+            ),
+            ImagesByVulnerabilityQuery(
+                "CVE-2020-8177",
+                [
+                    "sha256:fe3ca35038008b0eac0fa4e686bd072c9430000ab7d7853001bde5f5b8ccf60c"
+                ],
+            ),
+            ImagesByVulnerabilityQuery(
+                "CVE-2020-8177",
+                [],
+                ImagesByVulnerabilityQueryOptions(severity="High"),
+            ),
+            ImagesByVulnerabilityQuery(
+                "CVE-2020-8177",
+                [],
+                ImagesByVulnerabilityQueryOptions(namespace="debian:10"),
+            ),
+            ImagesByVulnerabilityQuery(
+                "CVE-2020-8177",
+                [],
+                ImagesByVulnerabilityQueryOptions(affected_package="gnupg2-2.0.22"),
+            ),
+            ImagesByVulnerabilityQuery("CVE-2020-0000", []),
+        ],
+    )
+    def test_query_by_vulnerability(
+        self, query_by_vuln_jsonschema, ingress_all_images, query
+    ):
+        get_image_resp = policy_engine_api.users.get_images_by_vulnerability(
+            vulnerability_id=query.vulnerability_id, **asdict(query.query_metadata)
+        )
+        assert get_image_resp.code <= 300
```

Comment:
Could you specify the exact response code expected here?

Reply:
fixed